oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/AuthorizationService.java

```diff
@@ -230,4 +231,18 @@ private boolean areClaimsValid(OIDCClaimsRequest claimsRequest, ClientRegistry c
     public String getExistingOrCreateNewPersistentSessionId(Map<String, String> headers) {
         return PersistentIdHelper.getExistingOrCreateNewPersistentSessionId(headers);
     }
+
+    public boolean isTestJourney(ClientID clientID, String emailAddress)
+            throws ClientNotFoundException {
+        var client =
+                dynamoClientService
+                        .getClient(clientID.toString())
+                        .orElseThrow(() -> new ClientNotFoundException(clientID.toString()));
+
+        return Optional.ofNullable(client)
+                .map(ClientRegistry::getTestClientEmailAllowlist)
```

Comment:
are we using this for smoke tests?

Reply:
We will adapt this for smoke tests right now, ahead of further changes to how we run them in future as discussed on Slack.